src/backend/TrafficCourtsApi/ticket-worker/Features/Emails/EmailSender.cs

```diff
@@ -51,12 +62,28 @@ public class EmailSender : IEmailSender
         private readonly IFluentEmail _email;
         private readonly ILogger<EmailSender> _logger;
         private readonly IEmailFilter _emailFilter;
+        private AsyncPolicyWrap<SendResponse> _policyWrap;
 
         public EmailSender(IFluentEmail email, ILogger<EmailSender> logger, IEmailFilter emailFilter)
         {
             _email = email ?? throw new ArgumentNullException(nameof(email));
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
             _emailFilter = emailFilter ?? throw new ArgumentNullException(nameof(emailFilter));
+
```

Comment:
Pulling some of the logic into extension methods and using built in or 3rd party back off implementations can make this simpler.   For example in another project, I implemented the following.  The `IsTransient` are extension methods on the exception type.   On `SendResponse` we can create an extension method that will return true if we should retry. Initially, it will be just checking the success, but later, we can check the error messages. We don't need to complicate core code with these concerns.  The implementation example, I was calling synchronous code. You should be able to substitute `RetryPolicy` for `AsyncRetryPolicy` in similar fashion.

```cs
        /// <summary>
        /// The retry policy that handles transient errors accessing EWS.
        /// </summary>
        private static RetryPolicy RetryPolicy = GetRetryPolicy();

        // ...

        private static RetryPolicy GetRetryPolicy()
        {
            var backoff = Backoff.ExponentialBackoff(TimeSpan.FromSeconds(1), 3);

            var policy = Policy
                .Handle<ServiceRemoteException>(ex => ex.IsTransient())
                .Or<ServiceRequestException>(ex => ex.IsTransient())
                .WaitAndRetry(backoff, OnRetry);

            return policy;
        }

        private static void OnRetry(Exception exception, TimeSpan delay, int retryCount, Context context)
        {
            log.Debug($"Operation {context.CorrelationId} failed and will be retried. Retry Count = {retryCount}, Retry Delay = {delay.TotalMilliseconds}ms", exception);
        }
```